### DISCUSSION TITLE
How to get preview by itself in sandpack?

### DISCUSSION THREAD

**Discussion by lancejpollard** [codesandbox/sandpack#1060] [Q&A]
I am looking to render the preview to PDF (from the server-side using puppeteer). So it needs a custom URL to load the preview, how do I do that?

Do I load the `<SandpackPreview>` by itself, with no layout, sort of like this (loaded at `https://preview.example.com/my-code` instead of just `/my-code`)?

```tsx
<SandpackProvider template="react">
  <SandpackPreview />
</SandpackProvider>
```

And/or do I need to hook up a subdomain to load the preview iframe securely? How should I accomplish this?

I am in Next.js if that matters.

**Comment by danilowoz** [comment]
It's a bit out of my expertise, I've never worked with puppeteer. But I remember when I implemented Playwright for e2e tests in Sandpack, and I faced a lot of challenges to make the bundler work: a) you need to make sure it can connect to an external domain b) find out when the iframe is loaded, evaluated and bundled everything.
